sunpy/net/helio/tests/test_helio.py

```diff
@@ -293,6 +294,20 @@ def test_client_search(client):
     res = client.search(a.Time(start, end), a.helio.TableName(table_name), a.helio.MaxRecords(10))
     assert len(res) == 10
 
+    with pytest.warns(SunpyUserWarning, match="Number of results is the same as current limit. "
+                      "It is possible your query has been truncated. "
+                      "If you want to change this, set `a.helio.MaxRecords` to a higher value."):
+        res = client.search(a.Time(start, end), a.helio.TableName(table_name), a.helio.MaxRecords(10))
```

Comment:
Why does this only get raised for this query?

Reply:
This query could return more than 10 records (I think it could return more than 500) so I think it's logical that it raises a warning when max record is equal to the length of the table.

Reply:
But isn't the previous line the same?

Reply:
I think it would be better to put it in a separate test called `test_client_search_warning(client)`, right?
And `test_client_search(client)` would be as before.

Reply:
Sure but I mean, isn't this test line the same as the one above?

Wouldn't both raise the warning, and if so, we don't need the second test?

Reply:
Yes it's the same, but it's for a different test purpose, that's why I see it's better to separate them.

Reply:
So why would we need to repeat it?

Reply:
Yeah it's not good.